pkg/scheduler/context.go

```diff
@@ -844,3 +888,16 @@ func (cc *ClusterContext) GetNode(nodeID, partitionName string) *objects.Node {
 	}
 	return partition.GetNode(nodeID)
 }
+
+func (cc *ClusterContext) checkForUnlimitedNodes() bool {
```

Comment:
This check should be per partition. A node is linked to one partition. The call for this check must be after the partition retrieval above, see previous comment

Reply:
We need to check for unlimited nodes cluster wide, no? If we moove it per partition, we might have an unlimited node in some other partition, than the onee we are checking. Is this acceptable?